src/utils/KML.js

```diff
@@ -237,7 +237,17 @@ const writeFeatures = (layer, featureProjection) => {
 
     const olStyle = new Style(newStyle);
     clone.setStyle(olStyle);
-    exportFeatures.push(clone);
+
+    if (
+      clone.getGeometry() instanceof Point &&
+      olStyle.getText() &&
+      !olStyle.getText().getText().length
+    ) {
+      // Don't save empty text feature
+      exportFeatures.push(new Feature());
```

Comment:
Here you save an empty feature. We just don'want to export features at all

Reply:
If there is one empty text, but several other features, we still want to export them, but without the empty one.
exportFeatures is an array of features to save in the kml

Reply:
Do ensure KML.writeFeatures() not to fail, I need to pass at least an empty feature.

Reply:
```
if (
     !( clone.getGeometry() instanceof Point &&
      olStyle.getText() &&
      !olStyle.getText().getText())
    ) {
      exportFeatures.push(clone);
    }
```

Reply:
That must be the final commit :angel: